src/util.js

```diff
@@ -54,7 +54,7 @@ const deserialize = (buffer) => {
 
   const data = pbn.Data == null ? Buffer.alloc(0) : pbn.Data
 
-  return new DAGNode(data, links, buffer.length)
+  return new DAGNode(data, links, buffer)
```

Comment:
This should be `DAGNode(data, links)` if the constructor is modified as suggested in my other review comment.

Reply:
good catch, comes from an intermediate version where I cached the entire `buffer` instead of just size but then I found git history where it was like that once but changed for memory reasons